File: PetMedicalCare/SoftPetBusiness/src/main/java/pe/edu/pucp/softpet/bo/DetalleDocumentoDePagoBo.java
package pe.edu.pucp.softpet.bo;

import java.util.ArrayList;
import pe.edu.pucp.softpet.dao.DetalleDocumentoDePagoDao;
import pe.edu.pucp.softpet.daoImp.DetalleDocumentoDePagoDaoImpl;
import pe.edu.pucp.softpet.daoImp.DocumentoDePagoDaoImpl;
import pe.edu.pucp.softpet.daoImp.ProductoDaoImpl;
import pe.edu.pucp.softpet.daoImp.ServicioDaoImpl;
import pe.edu.pucp.softpet.dto.facturacion.DetalleDocumentoPagoDto;

public class DetalleDocumentoDePagoBo {

    private DetalleDocumentoDePagoDao detalleDao;

    public DetalleDocumentoDePagoBo() {
        this.detalleDao = new DetalleDocumentoDePagoDaoImpl();
    }

    // Inserta un nuevo detalle de documento de pago
    public Integer insertar(int nroItem, String descripcion, int cantidad,
            double precioUnitario, double valorVenta,
           
            int documentoPagoId, int servicioId, int productoId) {

        DetalleDocumentoPagoDto detalle = new DetalleDocumentoPagoDto();
        detalle.setNroItem(nroItem);
        detalle.setDescripcion(descripcion.trim().toUpperCase());
        detalle.setCantidad(cantidad);
        detalle.setPrecioUnitario(precioUnitario);
        detalle.setValorVenta(valorVenta);
        detalle.setActivo(Boolean.TRUE);
        
      

        // Relaciones con otras entidades
        detalle.setDocumentoPago(new DocumentoDePagoDaoImpl().obtenerPorId(documentoPagoId));
        detalle.setServicio(new ServicioDaoImpl().obtenerPorId(servicioId));
        detalle.setProducto(new ProductoDaoImpl().obtenerPorId(productoId));

        return this.detalleDao.insertar(detalle);
    }

    // Modifica un detalle existente
    public Integer modificar(int detalleId, int nroItem, String descripcion, int cantidad,
            double precioUnitario, double valorVenta,
             double importeTotal,
            int documentoPagoId, int servicioId, int productoId, boolean esActivo) {

        DetalleDocumentoPagoDto detalle = new DetalleDocumentoPagoDto();
        detalle.setDddpId(detalleId);
        detalle.setNroItem(nroItem);
        detalle.setDescripcion(descripcion.trim().toUpperCase());
        detalle.setCantidad(cantidad);
         detalle.setPrecioUnitario(precioUnitario);
        detalle.setValorVenta(valorVenta);
        detalle.setActivo(esActivo);
        

        detalle.setDocumentoPago(new DocumentoDePagoDaoImpl().obtenerPorId(documentoPagoId));
        detalle.setServicio(new ServicioDaoImpl().obtenerPorId(servicioId));
        detalle.setProducto(new ProductoDaoImpl().obtenerPorId(productoId));

        return this.detalleDao.modificar(detalle);
    }

    // Elimina un detalle por su ID
    public Integer eliminar(int detalleId) {
        DetalleDocumentoPagoDto detalle = new DetalleDocumentoPagoDto();
        detalle.setDddpId(detalleId);
        return this.detalleDao.eliminar(detalle);
    }

    // Obtiene un detalle por ID
    public DetalleDocumentoPagoDto obtenerPorId(int detalleId) {
        return this.detalleDao.obtenerPorId(detalleId);
    }

    // Lista todos los detalles registrados
    public ArrayList<DetalleDocumentoPagoDto> listarTodos() {
        return this.detalleDao.listarTodos();
    }
}

File: PetMedicalCare/FrontEnd/SoftPetWA/SoftPetBussiness/SOAPBo/DocumentoPago/DetalleDocumentoDePagoBo.cs
using SoftPetBussiness.DetalleDocumentoDePagoClient; // Ajusta al namespace de tu Service Reference
using System;
using System.Collections.Generic;
using System.Linq;

namespace SoftPetBusiness
{
    public class DetalleDocumentoDePagoBO
    {
        private DetallesDocumentoDePagoClient clienteSOAP;

        public DetalleDocumentoDePagoBO()
        {
            this.clienteSOAP = new DetallesDocumentoDePagoClient();
        }
        /*
         *  detalleDocumentoPagoDto detalle = new detalleDocumentoPagoDto();
            detalle.activo = activo;
            detalle.cantidad = cantidad;
            detalle.descripcion = descripcion;
            detalle.documentoPago.documentoPagoId = documentoPagoId; //el listar solo me da el documento de pago id
            detalle.nroItem = nroItem;
            detalle.precioUnitario = precioUnitario;
            detalle.producto.productoId = productoId; //el listar solo me da el producto id
            detalle.servicio.servicioId = servicioId; //el listar solo me da el servicio id
            detalle.valorVenta = valorVenta;
         * 
         * 
         * 
         * 
         * 
         * 
         */
        // ===================================================
        //  INSERTAR (MISMA FIRMA QUE EL SERVICIO EN JAVA)
        //  Java:
        //  insertar_Detalle_Documento_De_Pago(int documentoPagoId, int servicioId,
        //                                      int productoId, int nroItem,
        //                                      String descripcion, int cantidad,
        //                                      double precioUnitario, double valorVenta, boolean activo)
        // ===================================================
        public int Insertar(
            int documentoPagoId,
            int servicioId,
            int productoId,
            int nroItem,
            string descripcion,
            int cantidad,
            double precioUnitario,
            double valorVenta,
            bool activo)
        {
            return this.clienteSOAP.insertar_Detalle_Documento_De_Pago(
                documentoPagoId,
                servicioId,
                productoId,
                nroItem,
                descripcion,
                cantidad,
                precioUnitario,
                valorVenta,
                activo
            );



        }

        // ===================================================
        //  MODIFICAR (MISMA FIRMA QUE EL SERVICIO EN JAVA)
        //  Java:
        //  modificar_Detalle_Documento_De_Pago(int ddpId, int documentoPagoId,
        //                                        int servicioId, int productoId,
        //                                        int nroItem, String descripcion,
        //                                        int cantidad, double precioUnitario,
        //                                        double valorVenta, boolean activo)
        // ===================================================
        public int Modificar(
            int ddpId,
            int documentoPagoId,
            int servicioId,
            int productoId,
            int nroItem,
            string descripcion,
            int cantidad,
            double precioUnitario,
            double valorVenta,
            bool activo)
        {
            return this.clienteSOAP.modificar_Detalle_Documento_De_Pago(
                ddpId,
                documentoPagoId,
                servicioId,
                productoId,
                nroItem,
                descripcion,
                cantidad,
                precioUnitario,
                valorVenta,
                activo
            );
        }

        // =========================
        //        ELIMINAR
        // =========================
        public int Eliminar(int ddpId)
        {
            return this.clienteSOAP.eliminar_Detalle_Documento_De_Pago(ddpId);
        }

        // =========================
        //      OBTENER POR ID
        // =========================
        public detalleDocumentoPagoDto ObtenerPorId(int ddpId)
        {
            return this.clienteSOAP.obtener_por_id(ddpId);
        }

        // =========================
        //       LISTAR TODOS
        // =========================
        public List<detalleDocumentoPagoDto> ListarTodos()
        {
            return this.clienteSOAP
                .listar_todos()
                .ToList<detalleDocumentoPagoDto>();
        }
    }
}
